### DISCUSSION TITLE
Assign a tag to a specific or multiple models

### DISCUSSION THREAD

**Discussion by mmfreitas** [netbox-community/netbox#7843] [Ideas]
Hello,

I would like to have the option to restrict a tag to one or to multiple models. For instance when creating a tag that I would only use in 'IPAM > IP Addresses' I don't need that tag nor do I want that tag associated to another model. Therefore would it be plausible to add a field of content types, similar to what already exists in 'Custom Fields'?

![image](https://user-images.githubusercontent.com/91552302/142023845-7b16bd2d-3136-405f-b7a1-814a49b05dfd.png)

I don´t know what code or what database changes this would require, so I'm openning this as a discussion instead of a feature request. What do you guys think?

**Comment by mtinberg** [comment]
I think there are probably a lot of different business rules about how data is input and related in Netbox for different organizations and that the core of Netbox should be kept as general as possible with per-organization sanity checking for whatever standards exist there implemented as Reports (and Scripts and Plugins) rather than building custom rules-checking infrastructure for each data model or feature that might benefit from it.  That's more flexible as you can write python code directly to implement your sanity checks, rather than needing some reduced functionality domain-specific-language to be created or additional database relations that affect performance and flexibility for everyone else.

It really depends on how many users would find this useful and how time consuming the code is to create and maintain long-term, if there weren't a lot of organizations who would need this feature, or it adds complexity then it's probably better as an external project to the core.  I personally think it's fine that tags don't have any namespace or restrictions and are entirely user-defined based on whatever policy your organization finds useful.  Usually if I am filtering a query with tags then I am operating on a specific data model, so if the tag is also applied to some other model that doesn't affect the results.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: MarcoMFreitas ***@***.***>
Sent: Tuesday, November 16, 2021 10:23 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Assign a tag to a specific or multiple models (Discussion #7843)


Hello,

I would like to have the option to restrict a tag to one or to multiple models. For instance when creating a tag that I would only use in 'IPAM > IP Addresses' I don't need that tag nor do I want that tag associated to another model. Therefore would it be plausible to add a field of content types, similar to what already exists in 'Custom Fields'?

[image]<https://user-images.githubusercontent.com/91552302/142023845-7b16bd2d-3136-405f-b7a1-814a49b05dfd.png>

I don´t know what code or what database changes this would require, so I'm openning this as a discussion instead of a feature request. What do you guys think?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/netbox-community/netbox/discussions/7843>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAS7UM6ZFHWEEAVVFJQGGPTUMKAPFANCNFSM5IEWBTBA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by candlerb** [reply]
I would find this useful.  Now that tags are predefined (#3703), it would be helpful to limit the set of tags presented to the user to the ones which make sense in the given context.

For example: some tags which relate to physical assets only make sense to apply to a DCIM > Device.
